atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java

```diff
@@ -258,9 +333,9 @@ private static SerializableTransactionManager create(
                 transactionManager,
                 kvs,
                 sweepRunner,
-                Suppliers.ofInstance(config.enableSweep()),
-                Suppliers.ofInstance(config.getSweepPauseMillis()),
-                Suppliers.ofInstance(getSweepBatchConfig(config)),
+                () -> MoreObjects.firstNonNull(runtimeConfig.get().enableSweep(), config.enableSweep()),
+                () -> MoreObjects.firstNonNull(runtimeConfig.get().getSweepPauseMillis(), config.getSweepPauseMillis()),
+                () -> getSweepBatchConfig(runtimeConfig.get(), config),
```

Comment:
Right now, all suppliers calls go through the `getSweepBatchConfig` or `ObjectUtils.firstNonNull()`, which assures that no null value will be used. While this behavior seems correct now, I could see it being broken in the future. We should probably add a unit test to this.